OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -101,7 +101,10 @@ public virtual void DoProduction(Actor self, ActorInfo producee, ExitInfo exitin
 
 		protected virtual Exit SelectExit(Actor self, ActorInfo producee, string productionType, Func<Exit, bool> p)
 		{
-			return self.RandomExitOrDefault(self.World, productionType, p);
+			if (!rp.IsValueCreated || rp.Value.Path.Count == 0)
```

Comment:
Why `rp.IsValueCreated`?

Reply:
Actually, I suppose I should remove the `Lazy` here. This trait actually does list `INotifyCreated` as interface, but only implements it because it inherits from `PausableConditionalTrait`. -.-

Reply:
We'll need to do something here because `rp.IsValueCreated` defeats the whole purpose of Lazy (which is to create the value when we need it)!

Reply:
ah nevermind, i missed that the second commit does this 😓